exonum-java-binding/core/src/main/java/com/exonum/binding/core/runtime/ServiceArtifactId.java

```diff
@@ -34,40 +34,50 @@
 public abstract class ServiceArtifactId {
 
   private static final String DELIMITER = ":";
-  private static final int NUM_FIELDS = 2;
+  private static final int KEEP_EMPTY = -1;
 
   /**
    * Returns the runtime id in which the service shall be deployed.
    */
   public abstract int getRuntimeId();
 
   /**
-   * Returns the full artifact name of this service (e.g., "com.acme:land-registry:1.2.0").
+   * Returns the artifact name of this service (e.g., "com.acme/land-registry").
    */
   public abstract String getName();
 
   /**
-   * Parses a service id in format "runtimeId:serviceName" as {@link #toString()} produces.
+   * Returns the artifact version of this service (e.g., "1.2.0").
+   */
+  public abstract String getVersion();
+
+  /**
+   * Parses a service id in format "runtimeId:serviceName:version" as {@link #toString()} produces.
    *
-   * @param serviceArtifactId a string in format "runtimeId:serviceName". Whitespace
+   * @param serviceArtifactId a string in format "runtimeId:serviceName:version". Whitespace
    *     characters, including preceding and trailing, are not allowed
    * @return a ServiceArtifactId with the given coordinates
    * @throws IllegalArgumentException if the format is not correct
    */
   public static ServiceArtifactId parseFrom(String serviceArtifactId) {
-    String[] coordinates = serviceArtifactId.split(DELIMITER, NUM_FIELDS);
+    JavaArtifactNames.checkNoForbiddenChars(serviceArtifactId);
+    String[] coordinates = serviceArtifactId.split(DELIMITER, KEEP_EMPTY);
+    checkArgument(coordinates.length == 3,
+        "Invalid artifact id (%s), must have 'runtimeId:groupId/artifactId:version' format",
```

Comment:
I'd say just 'runtimeId:serviceName:version' as we don't enforce groupId/artifactId